src/parser/SparqlParserHelpers.cpp

```diff
@@ -30,4 +36,73 @@ ParserAndVisitor::ParserAndVisitor(
     : ParserAndVisitor{std::move(input), disableSomeChecks} {
   visitor_.setPrefixMapManually(std::move(prefixes));
 }
+
+// _____________________________________________________________________________
+std::string ParserAndVisitor::unescapeUnicodeSequences(std::string input) {
+  std::u8string_view utf8View{std::bit_cast<char8_t*>(input.data()),
+                              input.size()};
+  std::string output;
+  size_t lastPos = 0;
+  UChar32 highSurrogate = 0;
+
+  auto throwError = [](bool condition, const std::string& message) {
+    if (!condition) {
+      throw InvalidSparqlQueryException{
+          absl::StrCat("Error in unicode escape sequence. ", message)};
+    }
+  };
+
+  constexpr size_t prefixLength = std::string_view{"\\U"}.size();
+
+  for (auto match :
+       ctre::search_all<R"(\\U[0-9A-Fa-f]{8}|\\u[0-9A-Fa-f]{4})">(utf8View)) {
```

Comment:
Currently you accept inputs like "<leadingSurrogate>something<trailingsurrogate>" whcih is not correct.